From 716860554eb49ab6cd5faa96d207effb24766a11 Mon Sep 17 00:00:00 2001
From: Sergio Lopez <slp@redhat.com>
Date: Wed, 11 Nov 2020 06:59:03 +0100
Subject: [PATCH 6/9] krunfw: setsockopt: lie about having set the option

Eventually we should inform the VMM so it can set the option on the
external context socket.

Signed-off-by: Sergio Lopez <slp@redhat.com>
---
 net/vmw_vsock/af_vsock.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/net/vmw_vsock/af_vsock.c b/net/vmw_vsock/af_vsock.c
index e342c68fdf0a..9e3c0e8bb59e 100644
--- a/net/vmw_vsock/af_vsock.c
+++ b/net/vmw_vsock/af_vsock.c
@@ -1660,7 +1660,10 @@ static int vsock_stream_setsockopt(struct socket *sock,
 	const struct vsock_transport *transport;
 	u64 val;
 
-	if (level != AF_VSOCK)
+	if (level == SOL_TCP) {
+		/* libkrun: lie about having set the option. */
+		return 0;
+	} else if (level != AF_VSOCK)
 		return -ENOPROTOOPT;
 
 #define COPY_IN(_v)                                       \
-- 
2.28.0

